scopes/lanes/ui/overview/menu/lanes-overview-menu.tsx

```diff
@@ -50,20 +52,26 @@ export function LanesOverviewMenu({ navigationSlot, widgetSlot, className }: Lan
     </div>
   );
 }
-
 function MenuNav({ navigationSlot }: { navigationSlot: LanesOrderedNavigationSlot }) {
-  const plugins = navigationSlot.toArray().sort(sortFn);
+  const plugins = flatten(
+    navigationSlot.toArray().map(([id, values]) => {
+      const flattenedValues = flatten(values).map((value) => ({ ...value, id }));
+      return flattenedValues;
+    })
+  ).sort(sortFn);
 
   return (
     <nav className={styles.navigation}>
-      {plugins.map(([id, menuItem]) => (
-        <TopBarNav key={id} {...menuItem.props} />
-      ))}
+      {plugins.map((menuItem, index) => {
+        const hidden = menuItem.hide?.();
+        if (hidden) return null;
+        return <TopBarNav key={`${menuItem.id}-${index}`} {...menuItem.props} />;
+      })}
     </nav>
   );
 }
 
-function sortFn([, { order: first }]: [string, NavPlugin], [, { order: second }]: [string, NavPlugin]) {
+function sortFn({ order: first }: LanesNavPlugin, { order: second }: LanesNavPlugin) {
```

Comment:
seems we have many usages for sorting slots based on the order. we should also extract to a component and try to reuse